### DISCUSSION TITLE
Plugin for .md file's stat?

### DISCUSSION THREAD

**Discussion by howesteve** [remarkjs/remark#855] [General] [answered]
Hello all,

I'm new to this. So I'm setting up a blog using sveltekit and mdsvex, which internally uses remark. 
So my question is, is there any plugin that returns the .md file's stats? I need to know when the .md file was created. For instance, this is returned from my blog setup:
``` js
{
  "default": {},
  "metadata": {
    "title": "Getting Started",
    "date": "9/9/2021",
    "readingTime": {
      "text": "3 min read",
      "minutes": 2.295,
      "time": 137700,
      "words": 459
    },
    "preview": "Thanks for checking out my blog template. It’s based on the blog I built for my own website and I hope this is a good starting point for you to start yours.",
    "previewHtml": "<p>Thanks for checking out my blog template. It’s based on the blog I built for my own <a href=\"https://mattjennings.io\" rel=\"nofollow\">website</a> and I hope this is a good starting point for you to start yours.</p>",
    "slug": "getting-started"
  }
}
```
As I said, I'm using sveltekit, in a SSR setup. If I try just to get the file name `fs.stat()` the file name will return an error:
``` html
Module "fs" has been externalized for browser compatibility and cannot be accessed in client code.
```
If someone wants to dig into, I'm running a setup similar to this: 
[https://github.com/mattjennings/sveltekit-blog-template/blob/main/src/lib/get-posts.js](https://github.com/mattjennings/sveltekit-blog-template/blob/main/src/lib/get-posts.js)

It would be useful to have the .md file's stat in metadata {} so that I could use stat.mdate | stat.mdate as defaults if a "date" header is not filled in that .md file.

Does anyone have a suggestion for this?

Thanks

**Comment by ChristianMurphy** [comment]
https://mdsvex.com/docs#remarkplugins--rehypeplugins can be used to pull in

* https://github.com/rehypejs/rehype-infer-description-meta
* https://github.com/rehypejs/rehype-infer-title-meta
* https://github.com/rehypejs/rehype-infer-reading-time-meta
* and/or https://github.com/rehypejs/rehype-meta

**Reply by howesteve** [reply]
Rehype is already being used in the setup shown above. Problem is, exactly what plugin injects file creation/modified time in meta{}?
None of these. rehype-meta **adds** custom data to the meta, not extracting. I didn't get your answer at all.

**Reply by wooorm** [reply]
https://github.com/unifiedjs/unified-infer-git-meta

**Comment by wooorm** [comment]
Most of your question in Svelte/mdsvex-specific btw — it looks like they have their own custom metadata place, and that they’re the ones that aren’t allowing you to use `fs`. Hence asking there might give you better results.

**Reply by howesteve** [reply]
Nope, it's Vite which is used by sveltekit as a preprocessor allowing SSR rendering. No node specific stuff in should be used in SSR rendering for obvious reasons.

**Reply by wooorm** [reply]
I’m not sure what you mean with “nope” because you seem to be asking how to use `fs.stat` but your setup doesn’t allow `fs.stat`? How could a remark plugin solve your setup.

**Reply by howesteve** [reply]
Not "my" setup of decision, it's vite which is used internally by sveltekit. Of course up to some point will read from file system, but to allow SSR the final rendered site cannot call node's fs. Any data should have been inserted in meta{} before rendering. That's how a plugin, if exists, could help. I think you're not getting it.

**Reply by wooorm** [reply]
> I think you're not getting it.

Indeed. That’s why I’m asking follow up questions. See [§ Asking quality questions in our support docs](https://github.com/remarkjs/.github/blob/main/support.md#asking-quality-questions) for some tips on how to ask questions.

---

> Any data should have been inserted in `meta{}` before rendering.

`meta{}` is not a unified/remark thing. We have our own metadata standards. How to add stuff to `meta{}` is a question better asked to the vite/sveltekit/mdsvex communities rather than us.

**Comment by howesteve** [comment, marked as answer]
Question was perfectly clear. Ok, I solved this using a simple plugin:

``` js
function addStat() {
  return (_, file) => {
    file.data.fm.stat = fs.statSync(file.filename);
  }
}
```

Thanks.
